src/tests/unit/tests/injected/path-snippet-controller.test.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { IMock, It, Mock, Times } from 'typemoq';
+import { PathSnippetStore } from '../../../../background/stores/path-snippet-store';
+import { PathSnippetStoreData } from '../../../../common/types/store-data/path-snippet-store-data';
+import { PathSnippetController } from '../../../../injected/path-snippet-controller';
+
+describe('InspectControllerTests', () => {
+    let pathSnippetStoreMock: IMock<PathSnippetStore>;
+    let pathSnippetStoreState: PathSnippetStoreData;
+    let addCorrespondingSnippetMock: IMock<(snippet: string) => void>;
+    let testObject: PathSnippetController;
+
+    beforeEach(() => {
+        pathSnippetStoreMock = Mock.ofType(PathSnippetStore);
+        addCorrespondingSnippetMock = Mock.ofInstance((snippet: string) => {});
+
+        testObject = new PathSnippetController(pathSnippetStoreMock.object, addCorrespondingSnippetMock.object);
+
+        pathSnippetStoreMock.setup(sm => sm.addChangedListener(It.isAny()));
```

Comment:
You may use a stronger parameter assertion here:
```typescript
import { isFunction } from 'lodash';
...
... sm.addChangedListener(It.is(isFunction)) ...
```

Also, by default, typemoq mocks use "Loose" behavior, which means, if you make a call you have not previously setup on the mock, there is no error being thrown at that point. 

In this case, you're setting up for a function that doesn't return anything but you probably want to make sure that it's being called. You can achieve this by adding the following function call after the `setup` call:
```typescript
.verifiable()
```

and then call this at the end of each related test:
```typescript
pathSnippetStoreMock.verifyAll();
```

Reply:
Gotcha! isFunction has been added in. I've been calling the verifyAll(); in the afterEach() method since I saw we were calling it with every test--does that accomplish the same thing? Thanks for the quick feedback!

Reply:
mmm `afterEach()` should be for clean up after the test, but `.verifyAll()` is part of the **AAA** test structure (**Arrange-Act-Assert**) so it should be part of the test function itself. If you happen to have more than one `.verifyAll()` at the end of each and every test, I'd recommend create a private function to run all those assertions there and then call said function from the tests.

I do know we still have `.verifyAll()` calls inside some `afterEach()` but after an engineering discussion we decided this was not the way we wanted to do the **Assert** part of the tests.

Reply:
oh got it! That makes sense I didn't realize afterEach() was more about cleanup. It looks like we call the two verifyAll() methods after each test, so I'll wrap them into a private function. Thanks for letting me know!

Reply:
I added the testObject.listenToStore() call to the private method as well because I saw we called it for every test. Let me know if you think it would be better left in the tests themselves for the sake of clarity!